File: moderate/UriComparison.java
/* CodeEval: URI Comparison
 * https://www.codeeval.com/open_challenges/80
 */

import java.io.*;
import java.net.URLDecoder;
import java.net.URLEncoder;

public class UriComparison {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            String[] urls = line.split(";");
            String url1 = urls[0].replaceAll("%(?![0-9a-fA-F]{2})", "%25");
            String url2 = urls[1].replaceAll("%(?![0-9a-fA-F]{2})", "%25");
            url1 = URLDecoder.decode(url1);
            url2 = URLDecoder.decode(url2);
            url1 = processUrl(url1);
            url2 = processUrl(url2);
            if (url1.equals(url2))
                System.out.println("True");
            else
                System.out.println("False");
        }
    }

    public static String processUrl(String url) {
        String newUrl = url.toLowerCase();
        newUrl = newUrl.replaceAll(":80", "");
        newUrl = URLEncoder.encode(newUrl);
        return newUrl;
    }
}

File: moderate/URIComparison.cs
/* CodeEval: URI Comparison
 * https://www.codeeval.com/open_challenges/80
 */

using System.IO;

class URIComparison
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            string[] urls = line.Trim().Split(';');
            string url1 =  DecodeUrl(urls[0]);
            string url2 =  DecodeUrl(urls[1]);
            if (url1 == url2)
                System.Console.WriteLine("True");
            else
                System.Console.WriteLine("False");
        }
    }

    static string DecodeUrl(string url)
    {
        string newUrl = url.ToLower();
        newUrl = newUrl.Replace(":80", "");
        newUrl = System.Uri.UnescapeDataString(newUrl); 
        return newUrl;
    }
}
